processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -107,10 +108,22 @@ public int getLimit()
       for (int i = 0; i < columns.size(); i++) {
         final OrderByColumnSpec columnSpec = columns.get(i);
 
+        if (aggAndPostAggNames.contains(columnSpec.getDimension())) {
+          sortingNeeded = true;
+          break;
+        }
+
+        final ValueType columnType = getOrderByType(columnSpec, dimensions);
+        final StringComparator naturalComparator;
+        if (columnType == ValueType.STRING) {
+          naturalComparator = StringComparators.LEXICOGRAPHIC;
+        } else {
```

Comment:
If outputType is `COMPLEX` or `null` this is not true. I know this is "impossible" but please check for it anyway.